src/main/java/cloud/eppo/EppoHttpClient.java

```diff
@@ -78,16 +68,19 @@ public void onResponse(Call call, Response response) {
                 if (response.isSuccessful()) {
                   log.debug("Fetch successful");
                   try {
-                    callback.onSuccess(response.body().string());
+                    future.complete(response.body().bytes());
                   } catch (IOException ex) {
-                    callback.onFailure("Failed to read response from URL " + httpUrl);
+                    future.completeExceptionally(
+                        new RuntimeException(
+                            "Failed to read response from URL {}" + request.url(), ex));
                   }
                 } else {
                   if (response.code() == HttpURLConnection.HTTP_FORBIDDEN) {
-                    callback.onFailure("Invalid API key");
+                    future.completeExceptionally(new RuntimeException("Invalid API key"));
```

Comment:
I know it was like this when you found it, but I wonder why do these additional checks with the asynchronous method but not the synchronous one.

Without looking too closely, it seems this code duplication shouldn't be needed and we could share most code with the synchronous `get()`, leveraging either calling it in a wrapped way, or some shared private base method.